lib/Backend/arm64/LowerMD.cpp

```diff
@@ -2680,7 +2680,7 @@ void LowererMD::ChangeToShift(IR::Instr *const instr, const bool needFlags)
     {
         // In the variable case, generate code to do the mask.
         IR::Opnd *const src2 = instr->UnlinkSrc2();
-        instr->SetSrc2(IR::RegOpnd::New(TyMachReg, func));
+        instr->SetSrc2(IR::RegOpnd::New(TyInt32, func));
```

Comment:
Should this be TyInt8 and then legalizer take care of it?

Reply:
While the result of the operation is guaranteed to fit into int8, the src1 operand of the AND is int32 so I think int32 dst makes more sense semantically?

Reply:
Fine